libvast/src/system/index.cpp

```diff
@@ -1010,8 +993,10 @@ index(index_actor::stateful_pointer<index_state> self,
             auto delta = std::chrono::steady_clock::now() - start;
             VAST_TRACEPOINT(query_meta_index, query_id.as_u64().first, total,
                             delta.count());
-            respond(query_id, detail::narrow<uint32_t>(total), scheduled);
-            rp.delegate(static_cast<index_actor>(self), query_id, scheduled);
+            rp.deliver(query_cursor{query_id, detail::narrow<uint32_t>(total),
+                                    scheduled});
+            caf::send_as(client, static_cast<index_actor>(self), query_id,
```

Comment:
Please add a comment above explaining to the future reader why we're not sending as the index but are spoofing the client instead.